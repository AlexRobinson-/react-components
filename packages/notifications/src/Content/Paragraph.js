import styled from 'styled-components';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts/dist/index.css';
import { retrieveTheme } from 'garden-react-theming';

const Paragraph = styled.div.attrs({
  className: classNames(CalloutStyles['c-callout__paragraph'])
})`
  ${props => retrieveTheme('notifications.paragraph', props)}
`;
/** @component */
export default Paragraph;

