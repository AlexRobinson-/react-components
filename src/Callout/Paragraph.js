import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

export default class Paragraph extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired
  };

  constructor(props, context) {
    super(props, context, {
      namespace: 'Callout',
      styles
    });
  }

  render() {
    const { children } = this.props;

    return (
      <p className={styles.paragraph}>
        {children}
      </p>
    );
  }
}
