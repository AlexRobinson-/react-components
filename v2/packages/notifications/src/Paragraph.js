import styled from 'styled-components';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts';
import { utils } from '@zendeskgarden/react-theming';

const Paragraph = styled.div.attrs({
  className: classNames(CalloutStyles['c-callout__paragraph'])
})`
  ${props => utils.retrieveTheme('callout.content', props)}
`;
/** @component */
export default Paragraph;

