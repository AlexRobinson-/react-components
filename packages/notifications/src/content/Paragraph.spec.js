import React from 'react';
import { shallow } from 'enzyme';
import Paragraph from './Paragraph';

it('should render default paragraph styling', () => {
  const wrapper = shallow(<Paragraph />)
  expect(wrapper).toMatchSnapshot();
});
