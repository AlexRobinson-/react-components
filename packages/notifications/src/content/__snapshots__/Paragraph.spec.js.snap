// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default paragraph styling 1`] = `
<div
  className="Paragraph-hHEPzZ c-callout__paragraph bHcknR"
/>
`;
